pkg/rancher-desktop/components/WSLProxy.vue

```diff
@@ -0,0 +1,141 @@
+<script lang="ts">
+import { Checkbox } from '@rancher/components';
+import _ from 'lodash';
+import Vue from 'vue';
+
+import RdFieldset from '@pkg/components/form/RdFieldset.vue';
+import { Settings } from '@pkg/config/settings';
+import { RecursiveTypes } from '@pkg/utils/typeUtils';
+
+import type { PropType } from 'vue';
+
+export default Vue.extend({
+  name: 'wsl-proxy',
+
+  components: { Checkbox, RdFieldset },
+
+  props: {
+    description: {
+      type:    String,
+      default: '',
+    },
+    emitChanges: {
+      type:    Boolean,
+      default: false,
+    },
+    firstRun: {
+      type:    Boolean,
+      default: false,
+    },
+    preferences: {
+      type:     Object as PropType<Settings>,
+      required: true,
+    },
+  },
+  computed: {
+    isFieldDisabled() {
+      return !(this.preferences.WSL.proxy.enabled || this.firstRun);
+    },
+  },
+  methods: {
+    onChange<P extends keyof RecursiveTypes<Settings>>(property: P, value: RecursiveTypes<Settings>[P]) {
+      if (this.emitChanges) {
+        const change = _.set({ WSL: { proxy: {} } }, property, value);
+
+        this.$emit('change', Object.assign(this.preferences.WSL.proxy, change.WSL.proxy));
+      } else {
+        this.$store.dispatch('preferences/updatePreferencesData', { property, value });
+      }
+    },
+  },
+});
+</script>
+
+<template>
+  <div class="preferences-body">
+    <rd-fieldset>
+      <section class="wsl-proxy">
+        <rd-fieldset v-if="!firstRun">
+          <checkbox
+            :label="t('proxy.windows.description', { }, true)"
+            :value="preferences.WSL.proxy.enabled"
+            @input="onChange('WSL.proxy.enabled', $event)"
+          />
+          <hr>
+        </rd-fieldset>
+        <h2 v-if="firstRun">
+          Configure Proxy Settings
+        </h2>
+        <rd-fieldset
+          class="wsl-proxy-fieldset"
+          :legend-text="t('proxy.windows.addressTitle', { }, true)"
+        >
+          <rd-fieldset
+            class="wsl-proxy-field"
+          >
```

Comment:
`rd-fieldset` is meant to be a grouping of like controls and labels, so I think we could reduce the amount of nesting used throughout.